cylc/flow/scheduler.py

```diff
@@ -815,14 +815,58 @@ def info_ping_task(self, task_id, exists_only=False):
         task_id = self.get_standardised_taskid(task_id)
         return self.pool.ping_task(task_id, exists_only)
 
+    def command_stop(
```

Comment:
As mentioned before (with the mutator resolver mapping), we should probably just create a new set of internal commands and not use `self.schd.server` methods as a mapping (since they will be deprecated).. And mark all the old ones for deprecation with CLI migration to `graphql`.

Reply:
The `self.schd.server` methods are up for deprecation but so is the `_mutation_mapper` method.

When we are finished mutations should have a sufficiently predictable interface that we can provide the mappings in a dictionary or else just use `getattr` directly onto the scheduler object so we won't need a programmatic mapping with hard-coded behaviours.

Reply:
Would be nice to signal back completion at some point.. I'm gonna suggest creating new sockets for each mutation at the UIS or move to ROUTER/DEALER (instead of REQ/REP) to fix the issue of doing two sends without a receive using the same socket error (feature of the REQ/REP pattern).. This will mean it might be ok for the client to wait around for a real result (instead of `"command queued"`).

Reply:
I was thinking rather than returning:

```python
(True, 'command queued')
```
We would return:

```python
(True, <subscriptionid>)
```

With `<subscriptionid>` being an identifier for the "queued command". The UI or CLI can then subscribe via the publisher URL to the topic `<subscriptionid>` to receive updates on the command.

We could potentially so something like this:

```python
class Scheduler:

    # ...

    def command_hold():
        yield 'Started'
        self.task_pool.hold()
        yield 'Suite Held'
```

Reply:
Saying that it would also be nice if we could do something for non-interactive use as well e.g:

```console
$ cylc stop <suite> 123 --await
$ echo $?
1
$ cylc cat-log <suite>
...
... Invalid cycle point 123 ...
```

Reply:
Both doable, although there's nothing stopping the `recv` from just waiting for a response (with timeout).. And we can always wrap it in a ThreadPoolExecutor (including socket creation) to hang around for the life of the request.

Reply:
One for the workshop?

Reply:
Yes, we should peg this as a topic under architecture "future work" (excuse the pun).